lib/yum/build.sh

```diff
@@ -33,13 +33,27 @@ rpmbuild_options=
 
 . /host/env.sh
 
-distribution=$(cut -d " " -f 1 /etc/system-release | tr "A-Z" "a-z")
-if grep -q Linux /etc/system-release; then
-  distribution_version=$(cut -d " " -f 4 /etc/system-release)
-elif grep -q Stream /etc/system-release; then
-  # ${version}-stream
-  distribution_version=$(cut -d " " -f 4 /etc/system-release)-$(cut -d " " -f 2 /etc/system-release | tr "[:upper:]" "[:lower:]")
+if [ -f /etc/os-release ]; then
+  . /etc/os-release
+  case $ID in
+    amzn)
+      # The ID is translated for repository layout
+      distribution=amazon
+      ;;
+    *)
+      distribution=$ID
+      ;;
+  esac
+  if grep -q Stream /etc/os-release; then
```

Comment:
Using grep is also OK.
Why don't you use `$NAME` for detecting `CentOS Stream`?

```console
$ . /etc/os-release
$ echo $NAME
CentOS Stream
```

Reply:
Thanks,
It should use $NAME in this context.